File: Ad-Hoc/2286/2286.c
#include <stdio.h>
 
int main() {
    int n, j1, j2, count = 1;
    char nome1[15], nome2[15];
    scanf("%d", &n);
    while (n != 0){
        scanf("%s", &nome1);
        scanf("%s", &nome2);
        printf("Teste %d\n", count);
        for (int i = 0; i<n; i++){
            scanf("%d %d", &j1, &j2);
            if ((j1+j2)%2 == 0) printf("%s\n", nome1);
            else printf("%s\n", nome2);
        }
        count++;
        printf("\n");
        scanf("%d", &n);
    }
    return 0;
}

File: Ad-Hoc/2286/2286.py
N = int(input())
count = 1
if N>=1 and N<=1000:
    while N!=0 and N<=1000:
        player1 = input().strip()
        player2 = input().strip()
        soma_rodada = 0
        vencedores_rodada = []
        vencedor_rodada = ""
        for x in range(0,N):
            par_impar = input().split()
            player1_numero = int(par_impar[0])
            player2_numero = int(par_impar[1])
            if player1_numero>=0 and player1_numero<=5 and player2_numero>=0 and player2_numero<=5:
                soma_rodada = player1_numero+player2_numero
            if soma_rodada%2 == 0:
                vencedor_rodada = player1
            else:
                vencedor_rodada = player2
            vencedores_rodada.append((vencedor_rodada))
        N = int(input())
        print(f"Teste {count}")
        for y in vencedores_rodada:
            print(y)
        print()
        count+=1